### DISCUSSION TITLE
Automatically enable custom ReqiuredActionProvider for all realms

### DISCUSSION THREAD

**Discussion by arahlf** [keycloak/keycloak#28482] [Q&A] [answered]
I'm creating a new custom ReqiuredActionProvider and was wondering if there's a good way to enable it for all realms on an instance. I have a handful of instances with dozens of realms (including one with over 100 realms), and was hoping to find a way through configuration to just have it be registered & enabled for all realms.

Worst case I suppose I could stitch something together and call REST APIs or something for each realm to do it but just wanted to see if there was an existing mechanism for this sort of thing.

Thanks

**Comment by codespearhead** [comment, marked as answer]
Realms are entirely independent of one another just like databases are isolated within the same Postgres instance, so it's very unlikely that the functionality to bridge realms will ever be introduced.

However, there is an ongoing discussion ( #23948 ) which I recommend you upvote, which boils down to the introduction of a new division within realms, called a tenant.

Until then, you can use the [Admin REST API](https://www.keycloak.org/docs/latest/server_development/#admin-rest-api) to enumerate the realms and add the RequiredActionProvider to all of them.
